core/trino-spi/src/main/java/io/trino/spi/security/AccessDeniedException.java

```diff
@@ -458,24 +458,24 @@ public static void denyRevokeTablePrivilege(String privilege, String tableName,
         throw new AccessDeniedException(format("Cannot revoke privilege %s on table %s%s", privilege, tableName, formatExtraInfo(extraInfo)));
     }
 
-    public static void denyShowRoles(String catalogName)
+    public static void denyShowRoles()
     {
-        throw new AccessDeniedException(format("Cannot show roles from catalog %s", catalogName));
+        throw new AccessDeniedException("Cannot show roles");
```

Comment:
Not sure there is much we can do here.  This is how most deny exceptions work.  In ConnectorAccessControl the default methods all deny, and since it is an interface there are is no way to have access to the catalog name.